jsonapi/unmarshal.go

```diff
@@ -10,19 +10,19 @@ import (
 // The UnmarshalIdentifier interface must be implemented to set the ID during
 // unmarshalling.
 type UnmarshalIdentifier interface {
-	SetID(string) error
+	SetID(Identifier) error
 }
 
 // The UnmarshalToOneRelations interface must be implemented to unmarshal
 // to-one relations.
 type UnmarshalToOneRelations interface {
-	SetToOneReferenceID(name, ID string) error
+	SetToOneReferenceID(name string, ID *RelationshipData) error
```

Comment:
why is RelationshipData? if we introduce Identifier struct? It's not handy to have different types for Identifiers

Reply:
replaced with *Identifier for ToOne and with []Identifier for ToMany